cli/popper/runner.py

```diff
@@ -18,48 +18,31 @@ class WorkflowRunner(object):
     # class variable that holds references to runner singletons
     runners = {}
 
-    def __init__(self, config_file=None, workspace_dir=os.getcwd(),
-                 reuse=False, engine_name='docker', dry_run=False, quiet=False,
-                 skip_pull=False, skip_clone=False):
+    def __init__(self, engine, resource_manager, config_file=None,
+                 workspace_dir=os.getcwd(), reuse=False, dry_run=False,
+                 quiet=False, skip_pull=False, skip_clone=False):
 
         # save all args in a member dictionary
-        self.config = DotMap(locals())
-        self.config.pop('self')
-        self.config.workspace_dir = os.path.realpath(workspace_dir)
-        self.config.wid = shake_256(workspace_dir.encode('utf-8')).hexdigest(4)
-
-        # cretate a repo object for the project
-        self.repo = scm.new_repo()
-        self.config.workspace_sha = scm.get_sha(self.repo)
-
-        if config_file:
-            # read options from config file
-            config_from_file = pu.load_config_file(config_file)
-
-            if hasattr(config_from_file, 'ENGINE'):
-                self.config.engine_options = config_from_file.ENGINE
-            if hasattr(config_from_file, 'RESOURCE_MANAGER'):
-                self.config.resman_options = config_from_file.RESOURCE_MANAGER
-
-        if not self.config.resman:
-            self.config.resman = 'host'
+        kwargs = locals()
+        kwargs.pop('self')
+        self.config = PopperConfig(**kwargs)
```

Comment:
really cool 😺

Reply:
we can remove the comment on line 25, or change it to something like "create a config object from the given arguments"

Reply:
Yeah, sorry.